testing/integration/tests/reentrancy_guard_test.rs

```diff
@@ -0,0 +1,309 @@
+use deposit_example::{deposit_actor::Method, WASM_BINARY as DEPOSIT_BINARY};
+use frc42_dispatch::method_hash;
+use fvm_integration_tests::{dummy::DummyExterns, tester::Tester};
+use fvm_ipld_blockstore::MemoryBlockstore;
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::{address::Address, bigint::Zero, econ::TokenAmount, MethodNum};
+
+mod common;
+
+pub const ATTACK_BINARY: Option<&[u8]> = Some(include_bytes!(
+    "../../../target/debug/wbuild/attack_example/attack_example.compact.wasm"
+));
+
+#[test]
+fn reeantrancy_attack_unguarded() {
+    let mut tester: Tester<MemoryBlockstore, DummyExterns> = common::create_tester();
+
+    let deposit_wasm_bin = DEPOSIT_BINARY.unwrap();
+    let deposit_actor_address = Address::new_id(10000);
+
+    let (_sender, ..) = common::init_actor_stateless_with_tester(
```

Comment:
You don't need _sender since it's not used anywhere
```suggestion
    common::init_actor_stateless_with_tester(
```